pkg/inject/virtualgateway_envoy.go

```diff
@@ -79,51 +61,62 @@ func (m *virtualGatewayEnvoyConfig) mutate(pod *corev1.Pod) error {
 	}
 
 	variables := m.buildTemplateVariables(pod)
-	envoyEnv, err := renderTemplate("vgenvoy", envoyVirtualGatewayEnvMap, variables)
-	if err != nil {
-		return err
-	}
+	envoy := pod.Spec.Containers[envoyIdx]
 
-	newEnvMap := map[string]string{}
-	err = json.Unmarshal([]byte(envoyEnv), &newEnvMap)
-	if err != nil {
-		return err
-	}
+	vg := fmt.Sprintf("mesh/%s/virtualGateway/%s", variables.MeshName, variables.VirtualGatewayOrNodeName)
+
+	envMap := map[string]string{}
+	updateEnvMapForEnvoy(variables, envMap, vg)
 
 	//we override the image to latest Envoy so customers do not have to manually manage
 	// envoy versions and let controller handle consistency versions across the mesh
 	if m.virtualGatewayImageOverride(pod) {
-		pod.Spec.Containers[envoyIdx].Image = m.mutatorConfig.sidecarImage
+		envoy.Image = m.mutatorConfig.sidecarImage
 	}
 
 	for idx, env := range pod.Spec.Containers[envoyIdx].Env {
-		if val, ok := newEnvMap[env.Name]; ok {
+		if val, ok := envMap[env.Name]; ok {
 			if val != env.Value {
-				pod.Spec.Containers[envoyIdx].Env[idx].Value = val
+				envoy.Env[idx].Value = val
 			}
-			delete(newEnvMap, env.Name)
+			delete(envMap, env.Name)
 		}
 	}
 
-	for name, value := range newEnvMap {
-		e := corev1.EnvVar{Name: name,
-			Value: value}
-		pod.Spec.Containers[envoyIdx].Env = append(pod.Spec.Containers[envoyIdx].Env, e)
+	for name, value := range envMap {
```

Comment:
Why can't we use the existing `getEnvoyEnv` function for this and append it to envoy.Env?

Reply:
Oh yes we can, I am doing that change